File: Algorithmen/Fakultät/Py/Fakultät.py
# Implementation des Fakultätsalgorithmus mit Iteration

import time


def Fakultät(n):
    if n == 0:
        return 1
    else:
        f = 1
        for i in range(1, int(n+1)):
            f *= i
        return f
    
    

n = float(input("Geben Sie eine natürliche Zahl ein: "))

t1 = time.time()

print(Fakultät(n))

print("Die Berechnung dauerte", int((time.time() - t1) * 1_000_000), "ns")

File: Algorithmen/Fakultät/cpp/Fakultät.cpp
#include <iostream>
#include <time.h>
#include <cmath>

using namespace std;

int main()
{
    double n;
    cout << "Geben Sie eine natürliche Zahl ein: ";
    cin >> n;

    clock_t t1;
    t1 = clock();

    if (n == 0)
    {
        cout << 1;
    }
    else
    {
        double f = 1;
        for (int i = 1; i <= n; i++)
        {
            f *= i;
        }
        cout << f;
    }

    cout << "\nDie Berechnung dauerte " << (clock() - t1) * (pow(10, 6)) / CLOCKS_PER_SEC << " ns";
    return 0;
}